source/common/buffer/buffer_impl.h

```diff
@@ -591,6 +591,11 @@ class OwnedImpl : public LibEventInstance {
    */
   virtual void appendSliceForTest(absl::string_view data);
 
+  // Does not implement watermarking.
+  void setWatermarks(uint32_t) override { ASSERT(false, "watermarks not implemented."); }
```

Comment:
I'm not a fan of having it in the base class API if it's not consistently implemented.  How about we just merge the two classes?

Reply:
Added TODO with longer term plan.